server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -565,13 +585,18 @@ List<CoordinatorDuty> makeIndexingServiceDuties()
 
   private List<CoordinatorDuty> makeMetadataStoreManagementDuties()
   {
-    return Arrays.asList(
-        new KillSupervisors(config, metadataManager.supervisors()),
-        new KillAuditLog(config, metadataManager.audit()),
-        new KillRules(config, metadataManager.rules()),
-        new KillDatasourceMetadata(config, metadataManager.indexer(), metadataManager.supervisors()),
-        new KillCompactionConfig(config, metadataManager.segments(), metadataManager.configs())
-    );
+    List<CoordinatorDuty> duties = new ArrayList<>();
+
+    duties.add(new KillSupervisors(config, metadataManager.supervisors()));
+    duties.add(new KillAuditLog(config, metadataManager.audit()));
+    duties.add(new KillRules(config, metadataManager.rules()));
+    duties.add(new KillDatasourceMetadata(config, metadataManager.indexer(), metadataManager.supervisors()));
+    duties.add(new KillCompactionConfig(config, metadataManager.segments(), metadataManager.configs()));
+
+    if (centralizedDatasourceSchemaConfig.isEnabled()) {
```

Comment:
I would advise using a different config to control cleanup of the schema, the same way that is done for the other metadata cleanup types. So, even if someone has centralized schema enabled, they may choose to turn off schema cleanup.

```
druid.coordinator.kill.segmentSchema.on=false (true by default)
druid.coordinator.kill.segmentSchema.period=PT1H
druid.coordinator.kill.segmentSchema.durationToRetain=PT6H
```
See any of the other metadata cleanup duties for reference.

Also, the new duty should extend `MetadataCleanupDuty`.